lib/core/api/alfresco-api/alfresco-api.http-client.spec.ts

```diff
@@ -0,0 +1,316 @@
+
+        });
+
+        it('should return parsed json object when responseType is json', (done) => {
+
+            const options: RequestOptions = {
+                path: '',
+                httpMethod: 'POST',
+                responseType: 'json'
+            };
+
+            angularHttpClient.request('http://example.com', options, securityOptions, emitters).then((res) => {
+                expect(res).toEqual(mockResponse);
+                done();
+            });
+
+            const req = controller.expectOne('http://example.com');
+            expect(req.request.method).toEqual('POST');
+
+            req.flush(mockResponse);
+
+        });
+
+        it('should emit unauthorized message for 401 request', (done) => {
+            const options: RequestOptions = {
+                path: '',
+                httpMethod: 'POST'
+            };
+
+            const spy = spyOn(emitter, 'emit').and.callThrough();
+
+            angularHttpClient.request('http://example.com', options, securityOptions, emitters).catch(() => {
+                expect(spy).toHaveBeenCalledWith('unauthorized');
+                done();
+            });
+
+            const req = controller.expectOne('http://example.com');
+            expect(req.request.method).toEqual('POST');
+
+            req.flush('<div></div>', { status: 401, statusText: 'unauthorized'});
+        });
+
+    });
+
+    describe('upload', () => {
+
+        afterEach(() => {
+            controller.verify();
+        });
+
+        it('should behave...', () => {
+            const requestOptions: RequestOptions = {
+                path: '/nodes/{nodeId}/children',
+                httpMethod: 'POST',
+                queryParams: {
+                    autoRename: true,
+                    include: 'allowableOperations',
+                    fields: null
+                },
+                formParams: {
+                    filedata: new File([], 'file.txt'),
+                    relativePath: '',
+                    include: ['allowableOperations'],
+                    renditions: 'doclib',
+                    autoRename: true,
+                    nodeType: 'cm:content'
+                },
+                bodyParam: {
+                    name: 'demo.txt',
+                    nodeType: 'cm:content',
+                    relativePath: '',
+                    newVersion: false,
+                    majorVersion: false,
+                    parentId: '-my-',
+                    path: ''
+                },
+                contentType: 'multipart/form-data',
+                accept: 'application/json',
+                returnType: null
+            };
+
+            angularHttpClient.request('http://example.com', requestOptions, securityOptions, emitters);
+            const req = controller.expectOne('http://example.com?autoRename=true&include=allowableOperations');
+            expect(req.request.method).toEqual('POST');
+
+            // filedata: (binary)
```

Comment:
these comments can be removed from the code

Reply:
oh, nice catch! thank you :)